crates/bevy_ui/src/render/mod.rs

```diff
@@ -347,12 +360,9 @@ pub fn extract_text_uinodes(
                 let rect = atlas.textures[index];
                 let atlas_size = Some(atlas.size);
 
-                // NOTE: Should match `bevy_text::text2d::extract_text2d_sprite`
                 let extracted_transform = global_transform.compute_matrix()
                     * Mat4::from_scale(Vec3::splat(scale_factor.recip()))
-                    * Mat4::from_translation(
-                        alignment_offset * scale_factor + text_glyph.position.extend(0.),
-                    );
+                    * Mat4::from_translation(alignment_offset + text_glyph.position.extend(0.));
```

Comment:
is it correct that the scale_factor should no longer be considered here?

Reply:
I *think* so. Scale factor is applied all over the place and it's very confusing though. I'll have another look

Reply:
I added an example `text_alignment` that cycles through different TextAlignments and scale factors and it definitely looks correct. I'm worried there are other edge cases but I think that even if there are some problems it's a definite improvement on before.

Reply:
I get it now, the whole alignment_offset doesn't need to be multiplied by the scale factor,  but the node_size does. I was missing that UiScale and scale_factor are different things. Everything should work now.